numba/cuda/tests/cudapy/test_atomics.py

```diff
@@ -715,6 +795,168 @@ def test_atomic_xor_global_2(self):
+                                                        orig - 1))))
+
+    def test_atomic_dec2(self):
+        rand_const = np.random.randint(32)
+        ary = np.random.randint(0, 32, size=32).astype(np.uint32).reshape(4, 8)
+        orig = ary.copy()
+
+        cuda_func = cuda.jit('void(uint32[:,:], uint32)')(atomic_dec2)
+        cuda_func[1, (4, 8)](ary, rand_const)
+
+        self.assertTrue(np.all(ary == np.where(orig == 0, rand_const,
+                                               np.where(orig > rand_const,
+                                                        rand_const,
+                                                        orig - 1))))
+
+    def test_atomic_dec3(self):
+        rand_const = np.random.randint(32)
+        ary = np.random.randint(0, 32, size=32).astype(np.uint32).reshape(4, 8)
+        orig = ary.copy()
+
+        cuda_func = cuda.jit('void(uint32[:,:], uint32)')(atomic_dec3)
+        cuda_func[1, (4, 8)](ary, rand_const)
+
+        self.assertTrue(np.all(ary == np.where(orig == 0, rand_const,
+                                               np.where(orig > rand_const,
+                                                        rand_const,
+                                                        orig - 1))))
+
+    def test_atomic_dec_global(self):
+        rand_const = np.random.randint(32)
+        idx = np.arange(32, dtype=np.uint32)
+        ary = np.random.randint(0, 32, size=32, dtype=np.uint32)
+        orig = ary.copy()
+        sig = 'void(uint32[:], uint32[:], uint32)'
+        cuda_func = cuda.jit(sig)(atomic_dec_global)
+        cuda_func[1, 32](idx, ary, rand_const)
+
+        self.assertTrue(np.all(ary == np.where(orig == 0, rand_const,
+                                               np.where(orig > rand_const,
+                                                        rand_const,
+                                                        orig - 1))))
+
+    def test_atomic_dec_global_2(self):
+        rand_const = np.random.randint(32)
+        ary = np.random.randint(0, 32, size=32).astype(np.uint32).reshape(4, 8)
+        orig = ary.copy()
+        cuda_func = cuda.jit('void(uint32[:,:], uint32)')(atomic_dec_global_2)
+        cuda_func[1, (4, 8)](ary, rand_const)
+
+        self.assertTrue(np.all(ary == np.where(orig == 0, rand_const,
+                                               np.where(orig > rand_const,
+                                                        rand_const,
+                                                        orig - 1))))
+
+    def test_atomic_exch(self):
+        rand_const = np.random.randint(50, 100, dtype=np.uint32)
+        ary = np.random.randint(0, 32, size=32).astype(np.uint32)
+        idx = np.arange(32, dtype=np.uint32)
+        gold = ary.copy()
+        gold.fill(rand_const)
+        cuda_func = cuda.jit('void(uint32[:], uint32[:], uint32)')(atomic_exch)
+        cuda_func[1, 32](ary, idx, rand_const)
+
+        np.testing.assert_equal(ary, gold)
```

Comment:
Is there a reason not to just check that `ary` is equal to `rand_const`, rather than creating a copy of `ary` and filling it with `rand_const`? (Also applies to the other `atomic_exch` tests)